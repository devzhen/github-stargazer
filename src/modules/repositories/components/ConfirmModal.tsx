import React from 'react';
import styled from 'styled-components/native';
import { Modal } from 'react-native';

import Button, { ButtonBackground } from '@components/Button';

interface ConfirmModalProps {
  submit: any;
  cancel: any;
  isVisible: boolean;
  isLoading: boolean;
  header: string;
  message: string;
  repoName: string;
}

const ConfirmModal = ({
  cancel,
  submit,
  isVisible,
  isLoading,
  header,
  message,
  repoName,
}: ConfirmModalProps) => (
  <>
    <Modal
      animationType="slide"
      transparent
      visible={isVisible}
      onRequestClose={() => {}}
    >
      <Container>
        <Header>{header}</Header>
        <Body>
          <Message>{message}</Message>
          <Repo>{repoName}</Repo>
        </Body>
        <Footer>
          <Button
            label="Cancel"
            onPress={cancel}
            bgColor={ButtonBackground.Gray}
          />
          <Spacer />
          <Button
            label="Ok"
            onPress={() => {
              if (isLoading) {
                return;
              }

              submit();
            }}
            disabled={isLoading}
            bgColor={ButtonBackground.Orange}
            loading={isLoading}
          />
        </Footer>
      </Container>
    </Modal>
    {isVisible && <Background />}
  </>
);

const Container = styled.View`
  margin-horizontal: 23px;
  margin-top: auto;
  margin-bottom: auto;
  padding: 30px;
  background-color: #fff;
  border-radius: 5px;
`;

const Footer = styled.View`
  flex-direction: row;
  width: 100%;
  height: 44px;
  padding-horizontal: 20px;
  flex-direction: row;
  justify-content: space-between;
  margin-top: 20px;
`;

const Header = styled.Text`
  font-size: 18px;
  font-weight: bold;
  color: #acacac;
  margin-bottom: 5px;
`;

const Body = styled.View`
  margin-top: 15px;
  margin-bottom: 0;
`;

const Spacer = styled.View`
  width: 20px;
`;

const Message = styled.Text`
  font-size: 16px;
  color: ${(props) => props.theme.colors.orange};
  margin-bottom: 15px;
`;

const Repo = styled.Text`
  font-size: 16px;
  color: ${(props) => props.theme.colors.lightBlack};
  margin-bottom: 15px;
`;

const Background = styled.View`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.5);
  z-index: 100000;
`;

export default ConfirmModal;
