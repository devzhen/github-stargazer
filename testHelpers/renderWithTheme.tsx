import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import 'jest-styled-components';

import theme from '@theme/index';

const renderWithTheme = (component: any) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

export default renderWithTheme;
