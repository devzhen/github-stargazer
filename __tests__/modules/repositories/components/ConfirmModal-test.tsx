import React from 'react';
import 'jest-styled-components';

import ConfirmModal from '@modules/repositories/components/ConfirmModal';

// @ts-ignore
import renderWithTheme from '@testHelpers/renderWithTheme';

describe('ConfirmModal repository component', () => {
  it('renders correctly, 1 case', () => {
    const tree = renderWithTheme(
      <ConfirmModal
        isVisible={true}
        cancel={() => {}}
        submit={() => {}}
        isLoading={true}
        repoName="test"
        header="test"
        message="test"
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly, 2 case', () => {
    const tree = renderWithTheme(
      <ConfirmModal
        isVisible={false}
        cancel={() => {}}
        submit={() => {}}
        isLoading={true}
        repoName="test"
        header="test"
        message="test"
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly, 3 case', () => {
    const tree = renderWithTheme(
      <ConfirmModal
        isVisible={false}
        cancel={() => {}}
        submit={() => {}}
        isLoading={false}
        repoName="test"
        header="test"
        message="test"
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
