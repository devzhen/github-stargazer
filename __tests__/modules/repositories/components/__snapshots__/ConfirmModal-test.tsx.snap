// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal repository component renders correctly, 1 case 1`] = `
Array [
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 5,
            "marginBottom": "auto",
            "marginHorizontal": 23,
            "marginTop": "auto",
            "paddingBottom": 30,
            "paddingLeft": 30,
            "paddingRight": 30,
            "paddingTop": 30,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#acacac",
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 5,
            },
          ]
        }
      >
        test
      </Text>
      <View
        style={
          Array [
            Object {
              "marginBottom": 0,
              "marginTop": 15,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#eb9e38",
                "fontSize": 16,
                "marginBottom": 15,
              },
            ]
          }
        >
          test
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#383838",
                "fontSize": 16,
                "marginBottom": 15,
              },
            ]
          }
        >
          test
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "height": 44,
              "justifyContent": "space-between",
              "marginTop": 20,
              "paddingHorizontal": 20,
              "width": "100%",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#d3d3d3",
              "borderRadius": 4,
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <Text
            labelSize={17}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 17,
                  "textAlign": "center",
                },
              ]
            }
          >
            Cancel
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "width": 20,
              },
            ]
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#eb9e38",
              "borderRadius": 4,
              "flexBasis": 0,
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "height": "100%",
              "justifyContent": "center",
              "opacity": 0.7,
            }
          }
        >
          <ActivityIndicator
            animating={true}
            color="#fff"
            hidesWhenStopped={true}
            size="small"
          />
        </View>
      </View>
    </View>
  </Modal>,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100000,
        },
      ]
    }
  />,
]
`;

exports[`ConfirmModal repository component renders correctly, 2 case 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 5,
          "marginBottom": "auto",
          "marginHorizontal": 23,
          "marginTop": "auto",
          "paddingBottom": 30,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 30,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#acacac",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 5,
          },
        ]
      }
    >
      test
    </Text>
    <View
      style={
        Array [
          Object {
            "marginBottom": 0,
            "marginTop": 15,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#eb9e38",
              "fontSize": 16,
              "marginBottom": 15,
            },
          ]
        }
      >
        test
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#383838",
              "fontSize": 16,
              "marginBottom": 15,
            },
          ]
        }
      >
        test
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "space-between",
            "marginTop": 20,
            "paddingHorizontal": 20,
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#d3d3d3",
            "borderRadius": 4,
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          labelSize={17}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 17,
                "textAlign": "center",
              },
            ]
          }
        >
          Cancel
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "width": 20,
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#eb9e38",
            "borderRadius": 4,
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "height": "100%",
            "justifyContent": "center",
            "opacity": 0.7,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#fff"
          hidesWhenStopped={true}
          size="small"
        />
      </View>
    </View>
  </View>
</Modal>
`;

exports[`ConfirmModal repository component renders correctly, 3 case 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 5,
          "marginBottom": "auto",
          "marginHorizontal": 23,
          "marginTop": "auto",
          "paddingBottom": 30,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 30,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#acacac",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 5,
          },
        ]
      }
    >
      test
    </Text>
    <View
      style={
        Array [
          Object {
            "marginBottom": 0,
            "marginTop": 15,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#eb9e38",
              "fontSize": 16,
              "marginBottom": 15,
            },
          ]
        }
      >
        test
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#383838",
              "fontSize": 16,
              "marginBottom": 15,
            },
          ]
        }
      >
        test
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "height": 44,
            "justifyContent": "space-between",
            "marginTop": 20,
            "paddingHorizontal": 20,
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#d3d3d3",
            "borderRadius": 4,
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          labelSize={17}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 17,
                "textAlign": "center",
              },
            ]
          }
        >
          Cancel
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "width": 20,
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#eb9e38",
            "borderRadius": 4,
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <Text
          labelSize={17}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 17,
                "textAlign": "center",
              },
            ]
          }
        >
          Ok
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
